### PR TITLE
create a single kms key for encryption inside eks module

### PR DESCRIPTION
**Pull request by j0hnmwangi**
# Description

This PR does the following:
 - modifies the module code to create only one kms key for encryption.
 - adds a `kms_key_id` variable to provide an external kms key.
 - adds new GitHub workflow and config files.

## Features/Fixes/Patches/Changes list:

Please check the [CHANGELOG.md](https://github.com/boldlink/terraform-aws-eks/blob/feature/create-single-key/CHANGELOG.md#310---2023-01-12)

### PR Owners Checklist:
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you checked there aren't open [Pull Requests](../../../pulls) of upstream or parallel code that will cause conflicts?
* [x] Have you updated the `CHANGELOG.md`?
* [x] Have you updated the `README.md`(s)?
* [x] OWNER: Does your submission pass?
    * [x] Pre-commit (attach logs/print screen to this PR)
    * [x] Checkov/terrascan (attach logs/print screen to this PR)
    * [x] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### PR Reviewers Checklists?
* [ ] Are all your comments/changes resolved?
* [ ] Are you happy with the OWNER checklists and additional information provided?
* [ ] Does your review tests pass?
    * [ ] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -83,6 +89,8 @@ fix: reference node iam arns inside the module for aws-auth configmap.
 
 [Unreleased]: https://github.com/boldlink/terraform-aws-eks/compare/3.0.2...HEAD
```
Comment (hugoalmeida264) [inline CHANGELOG.md]: This needs updating and a fun thing to also automate 💯 so add it to the backlog :D

File: .github/workflows/pre-commit.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - 'hotfixes/*'
+      - 'release/*'
+      - 'releases/*'
+      - 'chore/*'
+      - 'chores/*'
+      - 'automatic*'
+
+jobs:
+  pre-commit:
+    uses: boldlink/terraform-module-support/.github/workflows/pre-commit.yaml@1.0.0
```
Comment (hugoalmeida264) [inline .github/workflows/pre-commit.yaml:20]: should this not be updated to `...@main`?

File: .github/workflows/pr-labeler.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+name: PR Labeler
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+
+jobs:
+  pr-labeler:
+    uses: boldlink/terraform-module-support/.github/workflows/pr-labeler.yaml@1.0.0
```
Comment (hugoalmeida264) [inline .github/workflows/pr-labeler.yaml:10]: Is this the right version?

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - main
+jobs:
+  release:
+    uses: boldlink/terraform-module-support/.github/workflows/release.yaml@1.0.0
```
Comment (hugoalmeida264) [inline .github/workflows/release.yaml:9]: Is this the right version?

File: README.md [hunk 5]
```diff
@@ -1,4 +1,11 @@
-[![Build Status](https://github.com/boldlink/terraform-aws-eks/actions/workflows/pre-commit.yml/badge.svg)](https://github.com/boldlink/terraform-aws-eks/actions)
+[![License](https://img.shields.io/badge/License-Apache-blue.svg)](https://github.com/boldlink/terraform-aws-eks/blob/main/LICENSE)
```
Comment (hugoalmeida264) [inline README.md:1]: Please review the description of the module, it is lacking the detail on the module :)

File: examples/minimum/locals.tf [hunk 6]
```diff
@@ -4,7 +4,14 @@ locals {
...
   tags = {
-    environment        = "examples"
+    Environment        = "example"
+    Name               = local.cluster_name
     "user::CostCenter" = "terraform-registry"
+    InstanceScheduler  = true
+    Department         = "DevOps"
+    Project            = "Examples"
+    Owner              = "Boldlink"
+    LayerName          = "c700-eks-module-examples"
```
Comment (hugoalmeida264) [inline examples/minimum/locals.tf]: Should this be real layer names?

**Review by hugoalmeida264 — CHANGES_REQUESTED** [review]
See my comments and reach out if you have any questions 👍

File: .github/workflows/pre-commit.yaml (hunk 2 above)
Comment (j0hnmwangi) [reply .github/workflows/pre-commit.yaml:20]: Yes, the best place to update to main though is `terraform-module-template` repo, to also take effect in this repo during the next scheduled update

File: CHANGELOG.md (hunk 1 above)
Comment (j0hnmwangi) [reply CHANGELOG.md]: Alright

File: .github/workflows/release.yaml (hunk 4 above)
Comment (j0hnmwangi) [reply .github/workflows/release.yaml:9]: Yes, we however need to update the template repo to main

File: examples/minimum/locals.tf [hunk 7]
```diff
@@ -4,7 +4,14 @@ locals {
...
-    environment        = "examples"
+    Environment        = "example"
+    Name               = local.cluster_name
     "user::CostCenter" = "terraform-registry"
+    InstanceScheduler  = true
+    Department         = "DevOps"
+    Project            = "Examples"
+    Owner              = "Boldlink"
+    LayerName          = "cExample"
+    LayerId            = "c700"
```
Comment (hugoalmeida264) [inline examples/minimum/locals.tf]: Change the layer ID, match it to `cExample`

**Review by hugoalmeida264 — CHANGES_REQUESTED** [review]

**Review by hugoalmeida264 — APPROVED** [review]

**Pull request merged**